fpm/test/new_test/new_test.f90

```diff
@@ -110,11 +111,18 @@ program new_test
             write(*,*)'ERROR: internal error. unknown directory name ',trim(directories(i))
             stop 4
          end select
+         !! MSwindows has hidden files in it
          call list_files(trim(directories(i)), file_names,recurse=.true.)
+         if(allocated(fnames))deallocate(fnames)
+         allocate(character(len=0) :: fnames(0))
+         do j=1,size(file_names)
+           if(file_names(j)%s(1:1).eq.'.')cycle
```

Comment:
This might not work as intended: `list_files` returns filenames with the directory name prepended. Easy fix with `basename` to strip it before checking for `. `

Reply:
definitely differences and some unexpected differences. MSWindows output includes hidden files, Linux does not; was working around that rather than changing list_files but might want to make it consistent; etc. If it is not causing any problems this is probably going to take several iterations. I do not have a MSWIndows environment.

Reply:
Yes I agree, we should change `list_files` to be consistent.

Reply:
Well, I think I found
list_files shows hidden on MSWindows, does not on Linux. Not sure if that should be an option, if both should show hidden, or both no show hidden. Thinking default is not to show with optional parameter to show it?
mkdir on Linux silently will remake an existing file, on MSWindows remaking an existing directory is fatal. Since there is an exists and isdir function that makes it cleaner I was going to work around that. Apparently, git(1) is not in the path on the MSWindows system? If it is I do not know why call run(cd DIRNAME;git init) is failing. OK for this test; but if not there
I was looking at cargo and it has a switch to select which CVS to use and allows for none that I could add quickly (I do not need that but would like it to work just to verify it would but even if the problem is no git the -cvs switch might be good to add anyway.

Reply:
PS; might be something really simple. I have not kept up with all the development on the `build` command lately; was waiting till that settled down but `run` and `test` do not work with several of my fpm(1) packages. I am either asking for the wrong locations now (it worked and I think still does with running something made with h-fpm so something is different) or something is wrong.

Reply:
Are you able to point me to any of the packages that aren't building with f-fpm for me to have a look at?

Reply:
The main part builds, it is running it or building the test/ file that is not working as I hoped. The file structure is odd to accomodate my own build method, being on github with a build using make(1) and h-fpm(1) and trying to add f-fpm(1). I would prefer putting the app/ files back in a single directory which f-fpm supports (Thanks!) but that would currently break h-fpm(1). So the issues  for this one ([M_color](https://github.com/urbanjost/M_color)) are that
h-fpm allows me use #include to merge several modules into a file (f-fpm was not supporting remote dependencies and I needed something that would work with make(1) as well and that worked). So if for the time being I move the test directory to test.x and remove the test entries from fpm.toml it then builds the M_color module and builds the applications but puts them in build/*debug*/app where h-fpm put them in build/*debug*/PROGRAMS/. So they build, but the `run` command is looking for them where h-fpm put them; so I am not sure if it is `build` or `run` that needs changed to match. I think that resolving this one would fix several more complex ones, although they are similar. I have several non-typical layouts in the fpm registry that might make good tests and are easily accessible. I prefer your new layout capabilities but hate to break repositories with respect to h-fpm as there are users out there that might be testing -- but I think some of them "just" need the files re-arranged. So I pushed out M_color to show the problems I hit for now; hopefully without breaking anything. It is a simple module with multiple application demos and one unit test, but with a non-default name to accommodate the other methods.
Thanks for offering to take a look.

Reply:
Ah okay I understand your problem. Yes side-by-side use of `f-fpm` and `h-fpm` is problematic at the moment (I haven't been strict with keeping to `h-fpm` output convention), but it should still be possible to publish packages that are compatible with both (which I understand is important at this stage).

To answer your question, it is the `run` command that needs to change in this PR because currently `f-fpm` always puts programs in `build/gfortran_<release>/app` regardless of where the source file is located.

I've run into another issue while trying M_color with f-fpm; it's an already-known limitation of module dependency resolution in f-fpm whereby it does't check `include` files for module dependencies - not too difficult to fix, so I might address it this week.